### PR TITLE
Rust Cheat Sheet

### PR DESCRIPTION
**Pull request by sgp715**
## Type of Change

- [x] New Instant Answer
  - [x] Cheat Sheet
- [ ] Improvement
  - [ ] Bug fix
  - [ ] Enhancement
- [ ] Non–Instant Answer
  - [ ] Other (Role, Template, Test, Documentation, etc.)
## Description of new Instant Answer, or changes

Basic Rust Cheat Sheet

---

Forum Topic: 

IA Page: https://duck.co/ia/view/rust_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@sgp715 Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Rust Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/rust_cheat_sheet_mobile.png?rbZrelWVcs)

File: share/goodie/cheat_sheets/json/rust.json [hunk 1]
```diff
@@ -0,0 +1,130 @@
+{
+    "id": "rust_cheat_sheet",
+    "name": "Rust",
+    "description": "Basic Rust Cheat Sheet",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/rust.json]: lets make the description to describe the contents of cheat sheet

File: share/goodie/cheat_sheets/json/rust.json [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    "metadata": {
+        "sourceName": "The Rust Programming Book",
+        "sourceUrl" : "https://doc.rust-lang.org/stable/book/"
+    },
+
+    "aliases": [
+        ""
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/rust.json]: Add some aliases or remove this part

File: share/goodie/cheat_sheets/json/rust.json [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+		"Functions",
+		"Primitive Types"
+    ],
+
+    "sections": {
+		
+        "Variable Bindings": [
+            {
+                "key": "let x = 5;",
+                "val": "Bind the variable x to value 5."
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/rust.json]: Remove the periods from end of val fields

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Rust Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/rust_cheat_sheet_mobile.png?4lAAMSvW2N)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Rust Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/rust_cheat_sheet_mobile.png?F3ikan58KU)

**Comment by sgp715** [comment]
Ok, I made the changes. Thanks @jophab. This is my first PR, so let me know if I am doing anything wrong haha.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Rust Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/rust_cheat_sheet_mobile.png?tlZhCTH7WV)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Rust Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/rust_cheat_sheet_mobile.png?DNW6YavAaJ)

**Comment by pjhampton** [comment]
This PR is stale. Closing as a result. 

If you are looking for something more substantial to work through that will have a big impact for the users of DuckDuckGo, please check out the [DuckDuckHack Issues Table](https://duckduckhack.com/issues). 🚀  😄

**Pull request closed without merging**